cert-manager/namespaces/cert-manager/Deployment-cert-manager-webhook.yaml

```diff
@@ -1,65 +1,65 @@
-# Source: cert-manager/templates/webhook-deployment.yaml
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: cert-manager-webhook
-  namespace: "cert-manager"
   labels:
     app: webhook
-    app.kubernetes.io/name: webhook
+    app.kubernetes.io/component: webhook
     app.kubernetes.io/instance: cert-manager
-    app.kubernetes.io/managed-by: Tiller
-    helm.sh/chart: cert-manager-v0.13.1
+    app.kubernetes.io/name: webhook
+  name: cert-manager-webhook
+  namespace: cert-manager
 spec:
   replicas: 1
   selector:
     matchLabels:
-      app: webhook
-      app.kubernetes.io/name: webhook
+      app.kubernetes.io/component: webhook
       app.kubernetes.io/instance: cert-manager
-      app.kubernetes.io/managed-by: Tiller
+      app.kubernetes.io/name: webhook
   template:
     metadata:
       labels:
         app: webhook
-        app.kubernetes.io/name: webhook
+        app.kubernetes.io/component: webhook
         app.kubernetes.io/instance: cert-manager
-        app.kubernetes.io/managed-by: Tiller
-        helm.sh/chart: cert-manager-v0.13.1
-      annotations:
+        app.kubernetes.io/name: webhook
     spec:
-      serviceAccountName: cert-manager-webhook
       containers:
-        - name: cert-manager
-          image: "quay.io/jetstack/cert-manager-webhook:v0.13.1"
-          imagePullPolicy: IfNotPresent
-          args:
-          - --v=2
-          - --secure-port=10250
-          - --tls-cert-file=/certs/tls.crt
-          - --tls-private-key-file=/certs/tls.key
-          livenessProbe:
-            httpGet:
-              path: /livez
-              port: 6080
-              scheme: HTTP
-          readinessProbe:
-            httpGet:
-              path: /healthz
-              port: 6080
-              scheme: HTTP
-          env:
-          - name: POD_NAMESPACE
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.namespace
-          resources:
-            {}
-            
-          volumeMounts:
-          - name: certs
-            mountPath: /certs
-      volumes:
-      - name: certs
-        secret:
-          secretName: cert-manager-webhook-tls
+      - args:
+        - --v=2
+        - --secure-port=10250
+        - --dynamic-serving-ca-secret-namespace=$(POD_NAMESPACE)
+        - --dynamic-serving-ca-secret-name=cert-manager-webhook-ca
+        - --dynamic-serving-dns-names=cert-manager-webhook,cert-manager-webhook.cert-manager,cert-manager-webhook.cert-manager.svc
```

Comment:
Are the --tls-cert-file and --tls-private-key-file not necessary anymore?

Reply:
Also we're not mapping the "/certs" volume anymore, don't know if this is something that changed or if we need to be careful with this

Reply:
Nope, that's correct. cert-manager's webhook has to manage its own TLS to avoid circular dependencies